Platonus Tester/MainWindow.xaml.cs

```diff
@@ -463,14 +465,27 @@ private void FinishMenuItem_OnClick(object sender, RoutedEventArgs e)
             FinishTesting();
         }
 
-        public void OnSourceLoaded(SourceFile file)
+        public void OnSourceLoaded(object currentDispatcher)
         {
-            _sourcefile = file;
+            CurrentDispatcherFile sourcefile = (CurrentDispatcherFile)currentDispatcher;
+            Dispatcher targetDispatcher = sourcefile.Dispatcher as Dispatcher;
+            if (targetDispatcher == null)
+            {
+                return;
+            }
+
+            targetDispatcher.BeginInvoke(DispatcherPriority.Normal, new RunProcessDelegate(RunProcess), sourcefile.SourceFile);
+        }
+        public delegate void RunProcessDelegate(SourceFile sourcefile);
+
+        public void RunProcess(object sourcefile)
+        {
+
+            _sourcefile = (SourceFile)sourcefile;
```

Comment:
Here we set a value from anopther 'thread' to a private class field. That is bad pattern, and I got this a couple months ago. Please add a todo like:

```suggestion
           // TODO Maxim: here we set a value from another thread. This logic should be changed
            _sourcefile = (SourceFile)sourcefile;
```